src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs

```diff
@@ -815,40 +816,107 @@ internal static int GetHashCodeOrdinalIgnoreCase(ReadOnlySpan<char> value)
         // or are otherwise mitigated
         internal unsafe int GetNonRandomizedHashCode()
         {
+            uint hash1 = (5381 << 16) + 5381;
+            uint hash2 = hash1;
+            uint hash3 = hash1;
+            uint hash4 = hash1;
+
             fixed (char* src = &_firstChar)
             {
                 Debug.Assert(src[this.Length] == '\0', "src[this.Length] == '\\0'");
-                Debug.Assert(((int)src) % 4 == 0, "Managed string should start at 4 bytes boundary");
-
-                uint hash1 = (5381 << 16) + 5381;
-                uint hash2 = hash1;
+                Debug.Assert(((int) src) % 4 == 0, "Managed string should start at 4 bytes boundary");
 
-                uint* ptr = (uint*)src;
+                uint* ptr = (uint*) src;
                 int length = this.Length;
 
-                while (length > 2)
+                if(Vector128.IsHardwareAccelerated && length >= 2 * Vector128<ushort>.Count)
```

Comment:
```suggestion
                if (Vector128.IsHardwareAccelerated && length >= 2 * Vector128<ushort>.Count)
```
Nit: The repo coding style is to use space between `if` and `(`. (I am not commenting on all places.)